src/wcm.cpp

```diff
@@ -338,28 +339,124 @@ OptionWidget::OptionWidget(Option *option) : Gtk::Box(Gtk::ORIENTATION_HORIZONTA
+                    default_value))->length_ms;
+            easing_type = wf::option_type::from_string<wf::animation_description_t>(
+                wf_option->get_value_str())->easing_name;
+        }
 
         if (option->int_labels.empty())
         {
             auto spin_button = std::make_unique<Gtk::SpinButton>(
                 Gtk::Adjustment::create(value, option->data.min, option->data.max,
                     1));
             spin_button->signal_value_changed().connect(
-                [=, widget = spin_button.get()]
+                [=, widget = spin_button.get(), easing_type = easing_type]
                 {
-                    option->set_save(widget->get_value_as_int());
-                });
-            reset_button.signal_clicked().connect(
-                [=, widget = spin_button.get()]
-                {
-                    widget->set_value(std::get<int>(option->default_value));
+                    if (option->type == OPTION_TYPE_INT)
+                    {
+                        option->set_save(widget->get_value_as_int());
+                    } else
+                    {
+                        auto set_value =
+                            wf::option_type::from_string<wf::animation_description_t>(
+                                wf_option->get_value_str());
+                        auto easing =
+                            set_value.has_value() ? set_value->easing_name : std::get<std::string>(option->
+                                default_value);
+                        std::string animation_value = std::to_string(
+                            widget->get_value_as_int()) + "ms " + easing;
+                        option->set_save(animation_value);
+                    }
                 });
+            auto spinner = spin_button.get();
             pack_end(std::move(spin_button));
+            if (option->type == OPTION_TYPE_INT)
+            {
+                reset_button.signal_clicked().connect(
+                    [=, widget = spin_button.get()]
+                    {
+                        widget->set_value(std::get<int>(option->default_value));
+                    });
+            }
+
+            if (option->type == OPTION_TYPE_ANIMATION)
+            {
+                auto combo_box = std::make_unique<Gtk::ComboBoxText>();
+                std::string easing_types[] = {"linear", "circle", "sigmoid", "easeOutElastic"};
+                for (const auto & name : easing_types)
+                {
+                    combo_box->append(name);
+                }
+
+                int int_type = 0;
+                for (size_t i = 0; i < sizeof(easing_types); i++)
+                {
+                    if (easing_types[i] == easing_type)
+                    {
+                        int_type = i;
+                        break;
+                    }
+                }
+
+                combo_box->set_active(int_type);
```

Comment:
Consider using `set_active_text` here

Reply:
Any particular reason? Either way, it just looks like it would be the same thing with more code.